pqarrow/arrow.go

```diff
@@ -387,6 +389,56 @@ func writeColumnToArray(
 	w writer.ValueWriter,
 	dictionaryOnly bool,
 ) error {
+	if !repeated && dictionaryOnly {
+		// Check all the page indexes of the column chunk. If they are
+		// trustworthy and there is only one value contained in the column
+		// chunk, we can avoid reading any pages and construct a dictionary from
+		// the index values.
+		// TODO(asubiotto): This optimization can be applied at a finer
+		// granularity at the page level as well.
+		columnIndex := columnChunk.ColumnIndex()
+		columnType := columnChunk.Type()
+
+		globalMinValue := columnIndex.MinValue(0)
+		readPages := false
+		for pageIdx := 0; pageIdx < columnIndex.NumPages(); pageIdx++ {
+			if columnType.Length() == 0 {
+				// Variable-length datatype. The index can only be trusted if
+				// the size of the values is less than the column index size,
+				// since we cannot otherwise know if the index values are
+				// truncated.
+				if len(columnIndex.MinValue(pageIdx).Bytes()) >= dynparquet.ColumnIndexSize ||
```

Comment:
This is pretty confusing, but `.Bytes()` only works on values that are "binary" in parquet. Otherwise, this will panic. We should only perform this if that's the primitive type of the column. Otherwise we should just check once that the size of the type (eg. a uint64) fits into `ColumnIndexSize`.

Reply:
Gotcha, but these `Bytes` calls are within a `Length() == 0` check. My understanding is that all types that return `Length() == 0` are variable-width and so implicitly will return something meaningful from the `Bytes` call. In the `uint64` example, `Length` is 64 so we wouldn't get to this `Bytes` call. Is this a wrong assumption?